ts_src/block.ts

```diff
@@ -148,13 +148,20 @@ export class Block {
     return anyTxHasWitness(this.transactions!);
   }
 
-  byteLength(headersOnly?: boolean): number {
+  weight(): number {
+    const base = this.byteLength(false, false);
+    const total = this.byteLength(false, true);
+    return base * 3 + total;
+  }
+
+  byteLength(headersOnly?: boolean, allowWitness: boolean = true): number {
     if (headersOnly || !this.transactions) return 80;
 
     return (
       80 +
       varuint.encodingLength(this.transactions.length) +
-      this.transactions.reduce((a, x) => a + x.byteLength(), 0)
+      // @ts-ignore using the __byteLength private method on Transaction
+      this.transactions.reduce((a, x) => a + x.__byteLength(allowWitness), 0)
```

Comment:
Why don't you make public the `byteLength` method on transaction which accepts segwit option.

Reply:
I was just trying to not change existing APIs... but I guess we could make allowWitness default true.

Reply:
Check the new commit I just made.